src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -213,245 +332,20 @@ impl<'a> CodeGenerator<'a> {
-    ) -> CodeGenResult<()> {
-        let value = self.generate_field_value(isl_type_ref, modules, imports, code_gen_context)?;
-
-        tera_fields.push(Field {
-            name: {
-                match self.language {
-                    Language::Rust => field_name.to_case(Case::Snake),
-                    Language::Java => field_name.to_case(Case::Camel),
-                }
-            },
-            value,
-        });
-        Ok(())
-    }
-
-    /// Generates field value in a struct which represents a data type in codegen's programming language
-    fn generate_field_value(
-        &mut self,
-        isl_type_ref: &IslTypeRef,
-        modules: &mut Vec<String>,
-        imports: &mut Vec<Import>,
-        code_gen_context: &mut CodeGenContext,
-    ) -> CodeGenResult<String> {
-        Ok(match isl_type_ref {
-            IslTypeRef::Named(name, _) => {
-                if !self.is_built_in_type(name) {
-                    imports.push(Import {
-                        module_name: name.to_case(Case::Snake),
-                        type_name: name.to_case(Case::UpperCamel),
-                    });
-                }
-                let schema_type: IonSchemaType = name.into();
-                self.generate_sequence_field_value(
-                    schema_type.target_type(&self.language).to_string(),
-                    code_gen_context,
-                )
-            }
-            IslTypeRef::TypeImport(_, _) => {
-                unimplemented!("Imports in schema are not supported yet!");
-            }
-            IslTypeRef::Anonymous(type_def, _) => {
-                self.anonymous_type_counter += 1;
-                self.generate_data_model(modules, type_def)?;
-                let name = format!("AnonymousType{}", self.anonymous_type_counter);
-                imports.push(Import {
-                    module_name: name.to_case(Case::Snake),
-                    type_name: name.to_case(Case::UpperCamel),
-                });
-                self.generate_sequence_field_value(name, code_gen_context)
-            }
-        })
-    }
-
-    /// Generates an appropriately typed sequence in the target programming language to use as a field value
-    pub fn generate_sequence_field_value(
-        &mut self,
-        name: String,
-        code_gen_context: &mut CodeGenContext,
-    ) -> String {
-        if code_gen_context.data_model == Some(DataModel::Sequence) {
-            return match self.language {
-                Language::Rust => {
-                    format!("Vec<{}>", name)
-                }
-                Language::Java => {
-                    format!("ArrayList<{}>", name)
-                }
-            };
-        }
-        name
-    }
-
-    /// Generates Generates a read API for an abstract data type.
-    /// This adds statements for reading each the Ion value(s) that collectively represent the given abstract data type.
-    // TODO: add support for Java
-    fn generate_read_api(
```

Comment:
(🗺️ PR tour) Both `generate_read_api` and `generate_write_api` have been removed as the code here is moved to the template itself.